### PR TITLE
9068 Add encryption/locked validation and component tests to FileUploadPreview

### PR DESCRIPTION
**Pull request by argush3**
*Issue #:* /bcgov/entity#9068

*Description of changes:*

* Tests for covering validation and emitting of events for FileUploadPreview
* Addition of encryption and content lock validation checks

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

**Comment by argush3** [comment]
Was meaning to write some tests for file upload component but didn't quite get there before 9068 closed off.  But here they are...

File: src/mixins/document-mixin.ts [hunk 1]
```diff
@@ -4,7 +4,9 @@ import { Component, Vue } from 'vue-property-decorator'
 import { DocumentUpload } from '@/interfaces'
 import { PdfPageSize } from '@/enums'
 import pdfjsLib from 'pdfjs-dist/build/pdf'
-pdfjsLib.GlobalWorkerOptions.workerSrc = 'public/js/pdf.worker.min.js'
+import pdfjsWorker from 'pdfjs-dist/build/pdf.worker.entry'
+// Note: needs to be defined this way to work in tests
+pdfjsLib.GlobalWorkerOptions.workerSrc = pdfjsWorker
```
Comment (severinbeauvais) [inline src/mixins/document-mixin.ts:9]: Can `require()` be used instead?

**Review by severinbeauvais — APPROVED** [review]
Nice!

**Review by andrepestana-aot — APPROVED** [review]
Looks good. Do we have any tests for when a file is encrypted and should be rejected?

**Review by cameron-freshworks — APPROVED** [review]
Looks good!

**Comment by argush3** [comment]
> Looks good. Do we have any tests for when a file is encrypted and should be rejected?

The FE doesn't actually currently check for encryption.  i don't think pdf.js has any api methods to check for it either.

File: src/mixins/document-mixin.ts (hunk 1 above)
Comment (argush3) [reply src/mixins/document-mixin.ts:9]: will try

**Comment by severinbeauvais** [comment]
> Looks good. Do we have any tests for when a file is encrypted and should be rejected?

Or locked? (Since we might not be able to edit/stamp it ???)

**Comment by andrepestana-aot** [comment]
> > Looks good. Do we have any tests for when a file is encrypted and should be rejected?
> 
> The FE doesn't actually currently check for encryption. i don't think pdf.js has any api methods to check for it either.

I'm not sure if we can trust in this example but I did some tests and it seems to work: https://stackoverflow.com/q/52081345/16940959

**Comment by severinbeauvais** [comment]
> I'm not sure if we can trust in this example but I did some tests and it seems to work: https://stackoverflow.com/q/52081345/16940959

Do we have regular, encrypted (and also locked) PDFs to test with?

**Comment by andrepestana-aot** [comment]
> > I'm not sure if we can trust in this example but I did some tests and it seems to work: https://stackoverflow.com/q/52081345/16940959
> 
> Do we have regular, encrypted (and also locked) PDFs to test with?

You can create protected pdfs here: https://smallpdf.com/protect-pdf

File: src/mixins/document-mixin.ts (hunk 1 above)
Comment (argush3) [reply src/mixins/document-mixin.ts]: updated to use `require()`

File: jest.config.js [hunk 2]
```diff
@@ -7,6 +7,6 @@ module.exports = {
     }
   },
   preset: '@vue/cli-plugin-unit-jest/presets/typescript-and-babel',
-  transformIgnorePatterns: []
-
+  transformIgnorePatterns: [],
+  testTimeout: 30000 // need to extend timeout from default of 5000ms for FileUploadPreview component tests
```
Comment (severinbeauvais) [inline jest.config.js:11]: I think there's a way to override the timeout for just your test / test file. That would be preferable.

Comment (argush3) [reply jest.config.js:11]: ok, yeah that makes sense

File: src/mixins/document-mixin.ts [hunk 3]
```diff
@@ -3,8 +3,8 @@ import axios from '@/utils/axios-auth'
 import { Component, Vue } from 'vue-property-decorator'
 import { DocumentUpload } from '@/interfaces'
 import { PdfPageSize } from '@/enums'
-import pdfjsLib from 'pdfjs-dist/build/pdf'
-pdfjsLib.GlobalWorkerOptions.workerSrc = 'public/js/pdf.worker.min.js'
+const pdfjsLib = require('pdfjs-dist/build/pdf')
+pdfjsLib.GlobalWorkerOptions.workerSrc = require('pdfjs-dist/build/pdf.worker.entry')
```
Comment (severinbeauvais) [inline src/mixins/document-mixin.ts:7]: I'm sure you played with this, but does the import on line 6 need `require`, or can it follow the usual import syntax?

**Comment by argush3** [comment]
add check for encryption and also a test.  but wasn't able to figure out how to deal with locked files

**Review by severinbeauvais — COMMENTED** [review]
Great improvements!

So there's no need to check for a locked PDF? (Don't we need to edit it, to stamp it for example?)

**Comment by severinbeauvais** [comment]
> wasn't able to figure out how to deal with locked files

Doesn't matter if we don't need to edit them. But I'm just asking the questions - I don't really know :>

**Comment by argush3** [comment]
> > wasn't able to figure out how to deal with locked files
> 
> Doesn't matter if we don't need to edit them. But I'm just asking the questions - I don't really know :>

I think we need to check for locking.  I will do some more digging as we will need to edit the doc.  That was one check we missed on the BE I think.

File: jest.config.js (hunk 2 above)
Comment (argush3) [reply jest.config.js]: added timeouts to specific tests

File: src/mixins/document-mixin.ts (hunk 3 above)
Comment (argush3) [reply src/mixins/document-mixin.ts:7]: reverted line 6 to back what it was and it seems to work

**Comment by argush3** [comment]
> > > wasn't able to figure out how to deal with locked files
> > 
> > 
> > Doesn't matter if we don't need to edit them. But I'm just asking the questions - I don't really know :>
> 
> I think we need to check for locking. I will do some more digging as we will need to edit the doc. That was one check we missed on the BE I think.

Added content lock check validation.  Did some testing and although the BE doesn't explicitly check for this, it does return a "invalid file" error when the filing endpoint is hit as a result of an exception when trying to parse the file.

**Pull request merged**